contracts/NotifiableRewardAccumulator.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.19;
+
+import {IKwenta} from "./interfaces/IKwenta.sol";
+import {IStakingRewardsV2} from "./interfaces/IStakingRewardsV2.sol";
+import {ISupplySchedule} from "./interfaces/ISupplySchedule.sol";
+import {IRewardEscrowV2} from "./interfaces/IRewardEscrowV2.sol";
+
+contract NotifiableRewardAccumulator {
+
+    /// @notice kwenta interface
+    IKwenta internal immutable kwenta;
+
+    /// @notice rewards staking contract
+    IStakingRewardsV2 internal stakingRewardsV2;
+
+    /// @notice supply schedule contract
+    ISupplySchedule internal immutable supplySchedule;
+
+    /// @notice one time setter boolean
+    bool public stakingRewardsV2IsSet;
+
+    /// @notice Input address is 0
+    error InputAddress0();
```

Comment:
**Suggestion:** We use `ZeroAddress` for this error elsewhere, best to reuse this error for consistency.